haskell/test/SMT/Common/Gen.hs

```diff
@@ -0,0 +1,424 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | Generate the storage and entrypoint call
+module SMT.Common.Gen
+  ( genMkModelInput
+  , genXtzTransferType
+  , genTokenTransferType
+  ) where
+
+import Universum hiding (drop, swap)
+
+import Crypto.Random (drgNewSeed, seedFromInteger, withDRG)
+import qualified Data.Map as Map
+import Hedgehog.Gen.Tezos.Address (genAddress)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Lorentz hiding (not, cast, concat, get)
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Michelson.Test.Dummy
+import Hedgehog.Gen.Tezos.Crypto (genSecretKey)
+import Tezos.Address (Address(..), unsafeParseContractHash, mkKeyAddress)
+import Tezos.Crypto (SecretKey, toPublic, sign)
+import Util.Named ((.!))
+
+import Ligo.BaseDAO.Common.Types
+import Ligo.BaseDAO.Types
+import SMT.Model.BaseDAO.Types
+import SMT.Common.Types
+
+genMkModelInput :: SmtOption -> GeneratorT MkModelInput
+genMkModelInput option@SmtOption{..} = do
+
+  -- Initial
+  addrs <- genSecretKey
+    <&> (\secret -> (mkKeyAddress . toPublic $ secret, secret))
+    & vectorOf poolSize
+  startLevel <- Gen.integral (Range.constant 0 10)
+
+  modify (\s -> s { gsAddresses = addrs, gsLevel = startLevel, gsMkGenPropose = soMkPropose, gsMkCustomCalls = soMkCustomCalls })
+
+  -- Run Gen
+  mkContractCalls <- replicateM groupCallAmount genContractCallGroup
+
+  mkModelState <- genModelState option
+
+  pure $ \args@ModelInputArg{..} -> ModelInput
+    ( mkContractCalls <&> (\f -> f args)
+                      & concat
+    , mkModelState miaGuardianAddr miaGovAddr)
+  where
+    groupCallAmount = 2
+
+-- | Generate a list of BaseDAO contract calls that works together.
+genContractCallGroup :: GeneratorT (ModelInputArg -> [ModelCall])
+genContractCallGroup = do
+  proposingCalls <- genProposingProcess
+  mkRandCalls <- genRandomCalls
+  Gen.element [mkRandCalls, proposingCalls]
```

Comment:
There's a few other places where we're using `element` and could be simplified by using `choice` instead.

```suggestion
genContractCallGroup =
  Gen.choice [genProposingProcess, genRandomCalls]
```

Reply:
I've updated this.